webhooks/virtualmachine/mutation/virtualmachine_mutator_unit_test.go

```diff
@@ -292,111 +336,220 @@ func unitTestsMutating() {
-				Expect(err).ToNot(HaveOccurred())
-				Expect(mutated).To(BeTrue())
-				Expect(ctx.vm.Spec.ImageName).Should(Equal("vmi-123"))
+			When("spec.imageName matches a single cluster-scoped image", func() {
+				BeforeEach(func() {
+					ctx.vm.Spec.ImageName = clImg1Name
+				})
+				It("Should mutate Image but not ImageName", func() {
+					Expect(mutatedErr).ToNot(HaveOccurred())
+					Expect(wasMutated).To(BeTrue())
+					Expect(ctx.vm.Spec.Image).ToNot(BeNil())
+					Expect(ctx.vm.Spec.Image.Kind).To(Equal(cvmiKind))
+					Expect(ctx.vm.Spec.Image.Name).To(Equal(clImg1ID))
+					Expect(ctx.vm.Spec.ImageName).To(Equal(clImg1Name))
+				})
 			})
 		})
 
-		Context("When VM ImageName is set to a status name matching a single cluster scope image", func() {
-
+		When("spec.image is non-empty", func() {
 			BeforeEach(func() {
-				cvmi := builder.DummyClusterVirtualMachineImageA2("vmi-123")
-				cvmi.Status.Name = uniqueImageStatusName
-				Expect(ctx.Client.Create(ctx, cvmi)).To(Succeed())
-				ctx.vm.Spec.ImageName = uniqueImageStatusName
+				ctx.vm.Spec.Image = &vmopv1.VirtualMachineImageRef{
+					Kind: vmiKind,
+					Name: nsImg1ID,
+				}
 			})
-
-			It("Should mutate ImageName to the resource name of the cluster scope image", func() {
-				mutated, err := mutation.ResolveImageName(&ctx.WebhookRequestContext, ctx.Client, ctx.vm)
-				Expect(err).ToNot(HaveOccurred())
-				Expect(mutated).To(BeTrue())
-				Expect(ctx.vm.Spec.ImageName).Should(Equal("vmi-123"))
+			When("spec.imageName is vmi", func() {
+				BeforeEach(func() {
+					ctx.vm.Spec.ImageName = nsImg1ID
+				})
+				It("Should not mutate anything", func() {
+					Expect(mutatedErr).ToNot(HaveOccurred())
+					Expect(wasMutated).To(BeFalse())
+					Expect(ctx.vm.Spec.Image).ToNot(BeNil())
+					Expect(ctx.vm.Spec.Image.Kind).To(Equal(vmiKind))
+					Expect(ctx.vm.Spec.Image.Name).To(Equal(nsImg1ID))
+					Expect(ctx.vm.Spec.ImageName).To(Equal(nsImg1ID))
+				})
+			})
+			When("spec.imageName is display name", func() {
+				BeforeEach(func() {
+					ctx.vm.Spec.ImageName = nsImg1Name
+				})
+				It("Should not mutate anything", func() {
+					Expect(mutatedErr).ToNot(HaveOccurred())
+					Expect(wasMutated).To(BeFalse())
+					Expect(ctx.vm.Spec.Image).ToNot(BeNil())
+					Expect(ctx.vm.Spec.Image.Kind).To(Equal(vmiKind))
+					Expect(ctx.vm.Spec.Image.Name).To(Equal(nsImg1ID))
+					Expect(ctx.vm.Spec.ImageName).To(Equal(nsImg1Name))
+				})
+			})
+			When("spec.imageName is empty", func() {
+				BeforeEach(func() {
+					ctx.vm.Spec.ImageName = ""
+				})
+				It("Should not mutate anything", func() {
+					Expect(mutatedErr).ToNot(HaveOccurred())
+					Expect(wasMutated).To(BeFalse())
+					Expect(ctx.vm.Spec.Image).ToNot(BeNil())
+					Expect(ctx.vm.Spec.Image.Kind).To(Equal(vmiKind))
+					Expect(ctx.vm.Spec.Image.Name).To(Equal(nsImg1ID))
+					Expect(ctx.vm.Spec.ImageName).To(BeEmpty())
+				})
+			})
+			When("spec.imageName points to a different image", func() {
```

Comment:
cc @sreyasn @ericvmw this is where we test for the code at https://github.com/vmware-tanzu/vm-operator/pull/470/files#r1569229987